config/confighttp/clientinfohandler.go

```diff
@@ -50,7 +50,12 @@ func contextWithClient(req *http.Request, includeMetadata bool) context.Context
 	}
 
 	if includeMetadata {
-		cl.Metadata = client.NewMetadata(req.Header.Clone())
+		md := req.Header.Clone()
+		if len(md.Get("Host")) == 0 && req.Host != "" {
+			md.Add("Host", req.Host)
```

Comment:
Should "Host" be a const in the "client" package? Seems to be an important convention that we have.

Reply:
ok